### PR TITLE
Adding repository version commands

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: pulpcore/cli/file/repository_version.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import click
+
+from .repository import repository
+
+
+@repository.group()
+@click.pass_context
+def version(ctx):
+    if ctx.parent.params["repo_type"]:
+        ctx.obj.href_key = "file_file_repository_version_href"
```
Comment (mdellweg) [inline pulpcore/cli/file/repository_version.py:10]: I somehow feel this methodology has lost it's purpose here. We seem to need a thin ORM kind of layer here.

File: pulpcore/cli/file/repository_version.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    else:
+        raise NotImplementedError()
+
+
+@version.command()
+@click.option("--repository", required=True)
+@click.pass_context
+def list(ctx, repository):
+    search_result = ctx.obj.call(
+        "repositories_file_file_list", parameters={"name": repository, "limit": 1}
```
Comment (mdellweg) [inline pulpcore/cli/file/repository_version.py:22]: This should be dependent on the repository type.

File: pulpcore/cli/file/repository_version.py (hunk 1 above)
Comment (daviddavis) [reply pulpcore/cli/file/repository_version.py:10]: Agreed.

File: pulpcore/cli/file/repository_version.py (hunk 2 above)
Comment (daviddavis) [reply pulpcore/cli/file/repository_version.py:22]: I can update it. I meant to ask why we're checking for different repo types in file commands? Is that so these commands can be reused by other plugins?

Comment (mdellweg) [reply pulpcore/cli/file/repository_version.py:22]: Yeah, kind of. At the very least it was meant to show the technology for having different types.

**Pull request closed without merging**